src/main/java/pro/java/hw17/FileLogger.java

```diff
@@ -0,0 +1,153 @@
+package pro.java.hw17;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public class FileLogger {
+    private static final String RESET = "\u001B[0m";
+    private static final String RED = "\u001B[31m";
+    private static final String GREEN = "\u001B[32m";
+    private static final String ARROW = " ➜ ";
+    private final FileLoggerConfiguration config;
+    private PrintWriter writer;
+    private long currentFileSize;
+    private String currentFileName;
+
+    public FileLogger(FileLoggerConfiguration config) throws IOException {
+        this.config = config;
+        this.writer = config.isLogToConsole()
+                ? new PrintWriter(System.out)
+                : createFileWriter();
+    }
+
+    private PrintWriter createFileWriter() throws IOException {
+        this.currentFileName = config.getLogFileName();
+        this.currentFileSize = new File(currentFileName).length();
+        return new PrintWriter(new FileWriter(currentFileName, true));
+    }
+
+    private void checkFileSize() throws FileMaxSizeReachedException {
+        if (currentFileSize >= config.getMaxSize()) {
+            long fileSizeBeforeRotation = new File(currentFileName).length();
+
+            SimpleDateFormat dateFormat = new SimpleDateFormat(config.getLogFormat());
+            String timestamp = dateFormat.format(new Date());
+            String newFilePath = "Log_" + timestamp + ".txt";
+            try {
+                writer.close();
+
+                Path currentPath = Paths.get(currentFileName);
+                Path newPath = Paths.get(newFilePath);
+                Files.move(currentPath, newPath, StandardCopyOption.REPLACE_EXISTING);
+
+                writer = new PrintWriter(new FileWriter(currentFileName, true));
+                currentFileSize = 0;
+                currentFileName = newFilePath;
+
+                String errorMessage = "Maximum file size reached or exceeded!";
+                String limitMessage = "Log file size limit: " + config.getMaxSize() + " bytes";
+                String beforeRotationMessage = "Log file size before creating a new one: " + fileSizeBeforeRotation + " bytes";
+                String newLogFileMessage = "New log file created: " + currentFileName;
+                String newPathMessage = GREEN + "Path to the new log file: " + ARROW + newPath.toAbsolutePath() + RESET;
```

Comment:
What if move the code to construct these messages to separated method? It will be much more readable

Reply:
fixed